packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3734,15 +3739,15 @@ export class ContainerRuntime
 			}
 
 			const summaryContext =
-				lastAck === undefined
```

Comment:
should `lastAck` variable be gone? It's a bit weird that we have both lastAck & this.lastAckedSummaryContext and their values could be different.

Reply:
I kept it because of the validation in checkContinue(). I replaced that to validate that `this.lastAckedSummaryContext` is not updated in the middle of summarization. Can you please take a look if it looks correct.